src/librustc_trans/trans/mir/constant.rs

```diff
@@ -29,7 +30,17 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
         let ccx = bcx.ccx();
         let val = consts::trans_constval(bcx, cv, ty, bcx.fcx.param_substs);
         let val = if common::type_is_immediate(ccx, ty) {
-            OperandValue::Immediate(val)
+            let tty = type_of::type_of(ccx, ty);
+            if tty.is_aggregate() {
+                let lltemp = base::alloc_ty(bcx, ty, "__const_temp");
+                let store = build::Store(bcx, val, lltemp);
+                unsafe {
+                    llvm::LLVMSetAlignment(store, type_of::align_of(ccx, ty));
```

Comment:
nope that converts the target pointer and then the pointer and the type don't match